WebGPU API docs (#24202)
mdn/content: files/en-us/web/api/gpudevice/createtexture/index.md

    - `label` {{optional_inline}}
      - : A string providing a label that can be used to identify the object, for example in {{domxref("GPUError")}} messages or console warnings.
    - `mipLevelCount` {{optional_inline}}
      - : A number specifying the number of mip levels the texture will contain. If omitted, this defaults to 1.
    - `sampleCount` {{optional_inline}}
      - : A number specifying the texture's sample count. To be valid, the value must be 1 or 4. If omitted, this defaults to 1. A value higher than 1 indicates a multi-sampled texture.
    - `size`

      - : An object or array specifying the width, height, and depth/array layer count of the texture. The width value must always be specified, while the height and depth/array layer count values are optional and will default to 1 if omitted.

        What follows is a sample `size` array:

        ```js
        size: [16, 16, 2];
        ```

        The object equivalent would look like this:

        ```js
        size: {
          width: 16,
          height: 16,
          depthOrArrayLayers: 2
        }
        ```

    - `usage`

>       - : The {{glossary("bitwise flags")}} representing the allowed usages for the `GPUTexture`. The possible values can be found in the [`GPUTexture.usage` value table](/en-US/docs/Web/API/GPUTexture/usage#value).

Reviewer: ```suggestion
      - : The {{glossary("Bitwise_flags", "bitwise flags")}} representing the allowed usages for the `GPUTexture`. The possible values are in the [`GPUTexture.usage` value table](/en-US/docs/Web/API/GPUTexture/usage#value).
```

---
title: GPUDevice.createTexture()
slug: Web/API/GPUDevice/createTexture
page-type: web-api-instance-method
status:
  - experimental
browser-compat: api.GPUDevice.createTexture
---

{{APIRef("WebGPU API")}}{{SeeCompatTable}}

The **`createTexture()`** method of the
{{domxref("GPUDevice")}} interface creates a {{domxref("GPUTexture")}} in which to store texture data to use in GPU rendering operations.

## Syntax

```js-nolint
createTexture(descriptor)
```

### Parameters

- `descriptor`

  - : An object containing the following properties:

    - `dimension` {{optional_inline}}

      - : An enumerated value indicating the dimension level of the texture. Possible values are:

        - `"1d"`: The texture is one-dimensional.

>         - `"2d"`: The texture is an array of two-dimensional layers.

Reviewer 2: ```suggestion
        - `"2d"`: The texture is two-dimensional or an array of two-dimensional layers.
```

Author: updated
